glean/src/core/upload/index.ts

```diff
@@ -139,21 +151,34 @@ class PingUploader implements PingsDatabaseObserver {
     };
 
     const stringifiedBody = JSON.stringify(ping.payload);
+    // We prefer using `strToU8` instead of TextEncoder directly,
+    // because it will polyfill TextEncoder if it's not present in the environment.
+    // Environments that don't provide TextEncoder are IE and most importantly QML.
+    const encodedBody = strToU8(stringifiedBody);
+
+    let finalBody: string | Uint8Array;
+    let bodySizeInBytes: number;
     try {
-      const compressedBody = gzipSync(strToU8(stringifiedBody));
+      finalBody = gzipSync(encodedBody);
+      bodySizeInBytes = finalBody.length;
       headers["Content-Encoding"] = "gzip";
-      headers["Content-Length"] = compressedBody.length.toString();
-      return {
-        headers,
-        payload: compressedBody
-      };
+      headers["Content-Length"] = finalBody.length.toString();
```

Comment:
Let's reuse the variable. IMO that reads nicer and also ensures we're not missing it in case we ever change the code around it.

```suggestion
      headers["Content-Length"] = bodySizeInBytes.toString();
```

Reply:
Yeah and also that is a bugfix in itself, since `finalBody.length` is the number of characters in the string while `bidySizeInBytes` is, well, the body size in bytes aka what we actually want in this header.